zulipterminal/core.py

```diff
@@ -501,22 +516,33 @@ def check_for_prefixes(self, query: str) -> Tuple[str, List[List[str]]]:
         stream_prefix_present_flag = False
         final_search_text = ""
         new_narrow = []
-        # In the current version, the stream name and the topic name
-        # cannot have spaces within them
         for word in query.split(" "):
             if word.startswith("stream:"):
-                new_narrow.append(word.split(":"))
+                new_narrow.append(
+                    [
+                        word.split(":")[0],
```

Comment:
We're doing the same word.split operation a lot here, potentially repeatedly, and it likely will appear in other branches I suspect :)